src/main/storm/mesos/MesosNimbus.java

```diff
@@ -209,10 +345,6 @@ private OfferResources getResources(Offer offer, double executorCpu, double exec
         }
       }
     }
-
-    LOG.debug("Offer: " + offerToString(offer));
-    LOG.debug("Extracted resources: " + resources.toString());
-    return resources;
   }
 
   private List<WorkerSlot> toSlots(Offer offer, double cpu, double mem, boolean supervisorExists) {
```

Comment:
Within a given "cycle" -- where you're holding on to offers, are you combining them? A given slave can split up its resources.